pkg/agent/openflow/pipeline.go

```diff
@@ -1195,10 +1195,12 @@ func (f *featurePodConnectivity) l2ForwardCalcFlow(dstMAC net.HardwareAddr, ofPo
 }
 
 // l2ForwardOutputHairpinServiceFlow generates the flow to output the packet of hairpin Service connection with IN_PORT
-// action.
+// action. It excludes packets intended for packet-in handling by matching OutputToOFPortRegMark and 0 in PacketInOperationField.
 func (f *featureService) l2ForwardOutputHairpinServiceFlow() binding.Flow {
 	return OutputTable.ofTable.BuildFlow(priorityHigh).
 		Cookie(f.cookieAllocator.Request(f.category).Raw()).
+		MatchRegMark(OutputToOFPortRegMark).
```

Comment:
Is it necessary to keep both `OutputToOFPortRegMark` and `HairpinCTMark` ? I guess they are for the same purpose?

Reply:
My bad, I mean if `OutputToOFPortRegMark` and `MatchRegFieldWithValue(PacketInOperationField, 0)` are the same?